### ISSUE TITLE
clang 15 changed symbol mangling and might produce backwards incompatible binaries

### ISSUE THREAD

**Issue by isuruf** [conda-forge/clangdev-feedstock#213]
See https://releases.llvm.org/15.0.0/tools/clang/docs/ReleaseNotes.html#id5

> Fixed mangling of nested dependent names such as T::a::b, where T is a template parameter, to conform to the Itanium C++ ABI and be compatible with GCC. This breaks binary compatibility with code compiled with earlier versions of clang; use the -fclang-abi-compat=14 option to get the old mangling.

Mangled symbol changed from
```
__ZN6FFPACK8CharPolyIN6Givaro8Poly1DomINS1_7ModularIddvEENS1_5DenseEEEEERNT_7ElementERKS7_S9_mNS7_8Domain_t11Element_ptrEmRNS7_8Domain_t8RandIterENS_19FFPACK_CHARPOLY_TAGEm
```
to
```
__ZN6FFPACK8CharPolyIN6Givaro8Poly1DomINS1_7ModularIddvEENS1_5DenseEEEEERNT_7ElementERKS7_S9_mNS7_8Domain_t11Element_ptrEmRNSC_8RandIterENS_19FFPACK_CHARPOLY_TAGEm
```

**Comment by h-vetinari** [comment]
How big is the blast radius here? Do you want to use the compat ABI? Do we need a full-blown compiler migration just to get everything onto the new mangling?

**Comment by isuruf** [comment]
It's only one feedstock for now. Not sure how widespread it will be.

**Comment by h-vetinari** [comment]
Given that this issues doesn't seem to have a wide blast radius, shouldn't we simply recompile givaro and its dependents with LLVM 15?

**Comment by isuruf** [comment]
That will break old downstream packages depending on older ABI.

**Comment by h-vetinari** [comment]
That's what I meant with rebuilding its dependent packages (like a migration). Or do you mean consumers outside of conda-forge?

I don't think we should just keep blocking things like https://github.com/conda-forge/compilers-feedstock/pull/57 (or the next clang upgrade) indefinitely. The options I can imagine are:
* we "migrate" givaro to the new ABI
* do a full rebuild (with corresponding metadata) of conda-forge for the new ABI
* do nothing, but keep givaro on `clang-abi-compat=14` (until that option ceases to work)

Maybe I'm overlooking something?

**Comment by isuruf** [comment]
I sent PRs to keep givaro and its dependencies (through pkg-config file) with `clang-abi-compat=14`  after I opened this issue.

**Comment by h-vetinari** [comment]
Yes, I saw, which is why I said "keep givaro on...". Is this your preferred long-term solution? Does it remove your opposition to https://github.com/conda-forge/compilers-feedstock/pull/57?
